lib/mimic.ex

```diff
@@ -162,6 +162,42 @@ defmodule Mimic do
     )
   end
 
+  @doc """
+  Replace all public functions in `module` with public function in `mocking_module`.
+
+  If there's any public function in `module` that are not in `mocking_module`, it will raise if it's called
+
+  ## Arguments:
+
+    * `module` - The name of the module to stub.
+    * `mocking_module` - The name of the mocking module to stub the original module.
+
+  ## Raises:
+
+    * If `module` is not copied.
+    * If `function` is not called by the stubbing process.
+
+  ## Example
+      defmodule InverseCalculator do
+        def add(a, b), do: a - b
+      end
+
+      iex> Mimic.stubs_with(Calculator, InverseCalculator)
+      ...> Calculator.add(2, 4)
+      ...> -2
+
+  """
+  @spec stub(module()) :: module()
```

Comment:
Should we call it `stub_with` ?

Should this be `@spec stub_with` ?

Reply:
ah, definitely, let me change that

Reply:
changed @edgurgel